src/huggingface_hub/hf_api.py

```diff
@@ -3542,6 +3542,38 @@ def update_repo_visibility(
         hf_raise_for_status(r)
         return r.json()
 
+    @validate_hf_hub_args
+    def update_repo_settings(
+        self,
+        repo_id: str,
+        gated: Union[str, bool] = False,
```

Comment:
```suggestion
        gated: Literal["auto", "manual", False] = False,
```

(even more precise like this since `True` is not handled)